### PR TITLE
ref(models): Break up large file: `manager.py`

### PR DESCRIPTION
**Pull request by mgaeta**
This PR splits up a large file into one file per class. It is a redo of:
- #27791 which got too difficult to rebase
- #27871 which failed CI on getsentry

### CODE REVIEW

**Comment by ceorourke** [comment]
I can't see the test failures anymore on the other one, how is this different such that it will pass CI?

**Comment by mgaeta** [comment]
I'll have to manually test it and I'll add the results to the PR description.

File: mypy.ini [hunk 1]
```diff
@@ -66,6 +66,8 @@ no_implicit_reexport=True
 follow_imports = skip
 
 
+[mypy-celery.*]
```
Comment (ceorourke) [inline mypy.ini:69]: What does this and the line below it do?

Comment (untitaker) [reply mypy.ini:69]: mypy does not really understand dependencies for one reason or another (either the dependency not being installed while mypy runs or mypy being dumb)

For every dependency we use in a typed module we need to tell mypy to stop worrying as to whether the thing we're importing actually exists

**Review by untitaker — APPROVED** [review]

**Comment by mgaeta** [comment]
On getsentry I ran
```
pytest tests/getsentry/billing/utils/test_trial.py -svk test_plan_trial_start_subscription_integrity
```
And got `== 1 passed, 33 deselected in 12.36s ==`

**Pull request merged**